File: src/node/consumer/src/repository/models/torrent.ts
import { Table, Column, Model, HasMany, DataType } from 'sequelize-typescript';
import {ITorrentAttributes, ITorrentCreationAttributes} from "../interfaces/torrent_attributes";
import {Content} from "./content";
import {File} from "./file";
import {Subtitle} from "./subtitle";

@Table({modelName: 'torrent', timestamps: true})

export class Torrent extends Model<ITorrentAttributes, ITorrentCreationAttributes> {
    @Column({type: DataType.STRING(64), primaryKey: true})
    declare infoHash: string;
    
    @Column({type: DataType.STRING(32), allowNull: false})
    declare provider: string;
    
    @Column({type: DataType.STRING(512)})
    declare torrentId: string;
    
    @Column({type: DataType.STRING(512), allowNull: false})
    declare title: string;
    
    @Column({type: DataType.BIGINT})
    declare size: number;
    
    @Column({type: DataType.STRING(16), allowNull: false})
    declare type: string;
    
    @Column({type: DataType.DATE, allowNull: false})
    declare uploadDate: Date;
    
    @Column({type: DataType.SMALLINT})
    declare seeders: number;
    
    @Column({type: DataType.STRING(8000)})
    declare trackers: string;
    
    @Column({type: DataType.STRING(4096)})
    declare languages: string;
    
    @Column({type: DataType.STRING(16)})
    declare resolution: string;
    
    @Column({type: DataType.BOOLEAN, allowNull: false, defaultValue: false})
    declare reviewed: boolean;
    
    @Column({type: DataType.BOOLEAN, allowNull: false, defaultValue: false})
    declare opened: boolean;

    @HasMany(() => Content, { foreignKey: 'infoHash', constraints: false })
    contents?: Content[];

    @HasMany(() => File, { foreignKey: 'infoHash', constraints: false })
    files?: File[];
    
    subtitles?: Subtitle[];
}

File: src/shared/Models/Torrent.cs
namespace SharedContracts.Models;

public class Torrent
{
    public string? InfoHash { get; set; }
    public long? IngestedTorrentId { get; set; }
    public string? Provider { get; set; }
    public string? TorrentId { get; set; }
    public string? Title { get; set; }
    public long? Size { get; set; }
    public string? Type { get; set; }
    public string? UploadDate { get; set; }
    public int? Seeders { get; set; }
    public string? Trackers { get; set; }
    public string? Languages { get; set; }
    public string? Resolution { get; set; }
    public bool? Reviewed { get; set; }
    public bool? Opened { get; set; }
    public string? CreatedAt { get; set; }
    public string? UpdatedAt { get; set; }
}